### ISSUE TITLE
Reintroducing caching of regionprops

### ISSUE THREAD

**Issue by martinjanssens** [cloudsci/cloudmetrics#74]
In #70, we found our poor-man's caching of regionproperties to blow up memory up without bounds. In #70, we therefore removed the caching, with additional reasons that
- caching doesn't give much of a speed-up benefit anyway
- caching selectively (remembering a single scene) imposes an order on the user in which to compute metrics, sacrificing flexibility

If we find that computing the regionproperties every time is prohibitively slow, then let's pick this discussion back up again here.

**Comment by gmandorl** [comment]
Hi @martinjanssens, do you know how much time consuming is regionproperties? The benckmarks of #77 may be tested simultaneously on several metrics, thus caching regionproperties may be beneficial (?). Is it difficoult to implement the caching for a single `object_labels` as you suggested in #70 ?

**Comment by observingClouds** [comment]
Wouldn't we want to benchmark each metric from scratch without distorting the timings with cached results? With cached properties we might more easily run into issues with timing differences arising e.g. due to different orders of calling the metrics and it would also penalise the metrics that is called first if we want to compare timings across metrics.

**Comment by martinjanssens** [comment]
Yes, agree with @observingClouds that if we want metric speed to be part of the benchmarking, we should test them without cached properties, though I don't know that computation speed is among your current list of benchmarks, @gmandorl? For now, I would vote for trying out the benchmarking envisioned in #77 without cached regionprops, as usually (but this is a qualitative statement!) I haven't experienced the `get_regionprops` calls to consume a significant number of seconds (for fields of around 1000 pixels on a side and a relatively large number of objects, i.e. shallow convection). But if you find the benchmarking to really be slowed down by this, I'm happy to revisit it!

**Comment by gmandorl** [comment]
I did not consider benchmarking the speed. In that case, you're correct, we shouldn't cache the properties.
While implementing the benchmarks, I'll test their speed. If I find that `skmeasure.label` and `regionprops` are significantly time-consuming, I'll reopen the discussion here.
